lib/Elastica/Result.php

```diff
@@ -186,6 +186,21 @@ public function getExplanation()
     {
         return $this->getParam('_explanation');
     }
+    
+    /**
+     * Returns Document
+     * 
+     * @return \Elastica\Document
+     */
+    public function getDocument()
+    {
+        $doc = new \Elastica\Document();
+        $doc->setData($this->getSource());
+        $hit = $this->getHit();
+        unset($hit['_source']);
```

Comment:
http://php.net/manual/en/function.unset.php
> Note:
When using unset() on inaccessible object properties, the __unset() overloading method will be called, if declared.